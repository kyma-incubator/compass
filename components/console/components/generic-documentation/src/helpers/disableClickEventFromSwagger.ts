export function disableClickEventFromSwagger() {
  const beginOfSwaggerClickFunctionAsString = `for (var t = e.target; t && "A" !== t.tagName;)`;

  const originalEventListener = window.addEventListener;
  window.addEventListener = (
    type: string,
    fn: (...args: any[]) => any,
    options: any,
  ) => {
    if (type === 'click') {
      const fnAsString = fn.toString();
      if (fnAsString.includes(beginOfSwaggerClickFunctionAsString)) {
        return;
      }
    }
    return originalEventListener(type, fn, options);
  };
}
