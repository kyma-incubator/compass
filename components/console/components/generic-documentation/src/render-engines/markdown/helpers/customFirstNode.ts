import { Source } from '@kyma-project/documentation-component';
import { plugins } from '@kyma-project/dc-markdown-render-engine';
import { headingPrefix } from './headingPrefix';

type Header = plugins.Header;

export function customFirstNode(
  source: Source,
  toKebabCase: (str: string) => string,
): Header | undefined {
  const title: string =
    source.data && source.data.frontmatter && source.data.frontmatter.title;

  if (!title) {
    return;
  }

  return {
    title,
    id: toKebabCase(headingPrefix(source)),
    level: 'doc-title',
    source,
  };
}
