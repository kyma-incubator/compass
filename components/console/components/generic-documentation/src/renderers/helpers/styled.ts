import styled from 'styled-components';
import { ComboboxInput, Menu } from 'fundamental-react';

const typeColumnWidth = '6em';
const listItemColumnGap = '1em';
const listItemPadding = '2 * 10px';
export const ListItem = styled.span`
   {
    justify-items: start;
    display: grid;
    grid-gap: ${listItemColumnGap};
    grid-template-columns: ${typeColumnWidth} auto;
  }
`;

export const ApiTabHeader = styled.div`
   {
    display: grid;
    align-items: center;
    grid-template-columns: auto auto;
    grid-gap: 1em;
  }
`;

export const Combobox = styled(ComboboxInput)`
   {
    flex-shrink: 0;
    min-width: calc(
      ${props => props['data-max-list-chars'] + 'ch'} + ${typeColumnWidth} +
        ${listItemColumnGap} + ${listItemPadding}
    );
  }
`;

export const List = styled(Menu)`
   {
    max-height: 28em;
    overflow-y: auto;
  }
`;
