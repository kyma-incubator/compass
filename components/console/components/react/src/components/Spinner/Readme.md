```jsx
<Spinner size="40px" speed="2s" fontWeight="900" />
```

```jsx
<div>
    <Button primary>
        <Spinner color="#fff" />
    </Button>

    <Button secondary>
        <Spinner size="15px" color="#000" />
    </Button>

    <Button primary>
        <Spinner size="30px" color="#fff" padding="15px" />
    </Button>
</div>
```

