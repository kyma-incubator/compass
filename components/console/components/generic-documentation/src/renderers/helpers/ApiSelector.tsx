import React, { useState } from 'react';
import { Combobox, List, ListItem } from './styled';
import { Badge } from 'fundamental-react';
import {
  odataDefinition,
  asyncApiDefinition,
  markdownDefinition,
} from '../../constants';
import { Source } from '@kyma-project/documentation-component';

function getApiNameLength(s: Source) {
  return s.data && s.data.displayName ? s.data.displayName.length : 0;
}

const BadgeForType: React.FunctionComponent<{ type: string }> = ({ type }) => {
  let badgeType: 'success' | 'warning' | 'error' | undefined;

  if (odataDefinition.possibleTypes.includes(type)) {
    badgeType = 'warning';
  }

  if (asyncApiDefinition.possibleTypes.includes(type)) {
    badgeType = 'success';
  }

  if (markdownDefinition.possibleTypes.includes(type)) {
    badgeType = 'error';
  }

  return <Badge type={badgeType}>{type}</Badge>;
};

const ApiSelector: React.FunctionComponent<{
  sources: Source[];
  onApiSelect: (api: Source) => void;
  selectedApi: Source;
}> = ({ sources, onApiSelect, selectedApi }) => {
  const [searchText, setSearchText] = useState('');

  const filteredSources = sources.filter(
    (s: Source) =>
      (s.data &&
        s.data.displayName &&
        s.data.displayName.toUpperCase().includes(searchText.toUpperCase())) ||
      s.type.includes(searchText),
  );

  function handleInputChange(e: React.ChangeEvent<HTMLInputElement>) {
    setSearchText(e.target.value);
  }

  const maxApiNameLength = filteredSources.length
    ? getApiNameLength(
        filteredSources.reduce((prev, current) =>
          getApiNameLength(current) > getApiNameLength(prev) ? current : prev,
        ),
      )
    : 0;

  return (
    <Combobox
      onClick={(e: React.MouseEvent<HTMLElement>) => {
        const a = e.target as HTMLElement; // not sure why but it's needed, thank you typescript!
        if (a.tagName === 'INPUT' || a.tagName === 'BUTTON') {
          e.stopPropagation(); // avoid closing the dropdown due to the "opening" click âˆž
        }
      }}
      data-max-list-chars={maxApiNameLength}
      menu={
        <List>
          {filteredSources.map((s: Source, id) => (
            <a
              aria-label="api-"
              href="#"
              onClick={e => onApiSelect(s)}
              className="fd-menu__item"
              key={(s.data && s.data.displayName) || s.rawContent}
            >
              <ListItem>
                <BadgeForType type={s.type} />
                {s.data && s.data.displayName}
              </ListItem>
            </a>
          ))}
        </List>
      }
      placeholder={
        (selectedApi && selectedApi.data && selectedApi.data.displayName) ||
        (selectedApi && selectedApi.type) ||
        'Select API'
      }
      inputProps={{ onChange: handleInputChange }}
    />
  );
};

export default ApiSelector;
