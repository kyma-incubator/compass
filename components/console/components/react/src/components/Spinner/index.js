import React from 'react';
import styled from 'styled-components';

const SpinnerWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  width: 75px;
  height: 75px;
`;

const Spinner = () => (
  <SpinnerWrapper>
    <div className="fd-spinner" aria-hidden="false" aria-label="Loading">
      <div />
    </div>
  </SpinnerWrapper>
);

export default Spinner;
