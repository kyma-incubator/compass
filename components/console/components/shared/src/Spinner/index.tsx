import React from 'react';
import { SpinnerWrapper } from './styled';

export const Spinner: React.FunctionComponent = () => (
  <SpinnerWrapper>
    <div className="fd-spinner" aria-hidden="false" aria-label="Loading">
      <div />
    </div>
  </SpinnerWrapper>
);
