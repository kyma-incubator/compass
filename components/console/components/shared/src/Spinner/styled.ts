import styled from 'styled-components';

export const SpinnerWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  width: 75px;
  height: 75px;
`;
