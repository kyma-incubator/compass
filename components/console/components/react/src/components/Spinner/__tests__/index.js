import React from 'react';
import Spinner from '../index';

import { renderWithTheme } from 'test_setup/helpers';

describe('Spinner Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<Spinner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
