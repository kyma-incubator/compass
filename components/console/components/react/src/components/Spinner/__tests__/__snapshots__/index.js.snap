// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Element renders correctly 1`] = `
<div
  className="sc-bdVaJa JBcwN"
/>
`;
